src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,465 @@
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: uniform_distribution_rvs
+    public :: uniform_distribution_pdf
+    public :: uniform_distribution_cdf
+    public :: shuffle
+
+    interface uniform_distribution_rvs
+    !! Version experimental
+    !!
+    !! Get uniformly distributed random variate for integer, real and complex
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        module procedure unif_dist_rvs_0_rsp                 ! 0 dummy variable
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_1_${t1[0]}$${k1}$     ! 1 dummy variable
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_${t1[0]}$${k1}$       ! 2 dummy variables
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_array_${t1[0]}$${k1}$ ! 3 dummy variables
+        #:endfor
+    end interface uniform_distribution_rvs
+
+    interface uniform_distribution_pdf
+    !! Version experiment
+    !!
+    !! Get uniform distribution probability density (pdf) for integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_pdf_${t1[0]}$${k1}$
+        #:endfor
+    end interface uniform_distribution_pdf
+
+    interface uniform_distribution_cdf
+    !! Version experimental
+    !!
+    !! Get uniform distribution cumulative distribution function (cdf) for integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+    !!
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_cdf_${t1[0]}$${k1}$
+        #:endfor
+    end interface uniform_distribution_cdf
+
+    interface shuffle
+    !! Version experimental
+    !!
+    !! Fisher-Yates shuffle algorithm for a rank one array of integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+    !!
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure shuffle_${t1[0]}$${k1}$
+        #:endfor
+    end interface shuffle
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    impure elemental function unif_dist_rvs_1_${t1[0]}$${k1}$(scale) result(res)
+    ! Uniformly distributed integer in [0, scale]
+    ! Bitmask with rejection
+    ! https://www.pcg-random.org/posts/bounded-rands.html
+    !
+    ! Fortran 90 translated from c by Jim-215-fisher
+        ${t1}$, intent(in) :: scale
+        ${t1}$ ::  res, u, mask, n
+        integer :: zeros, bits_left, bits
+
+        n = scale
```

Comment:
Why is `scale` copied in `n`? It could be avoided,right?

Reply:
The usage of variable 'scale' is purely for consistency and self-explanatory.  Yes, it could be avoided.